### ISSUE TITLE
Rework Contributors/Administrators Permissions Dialog

### ISSUE THREAD

**Issue by hmmwhatsthisdo** [jakubgarfield/Bonobo-Git-Server#500]
Currently, the Contributors/Administrators permissions dialog is two checkbox-lists with identical contents.

It would likely be better if these two lists were zipped together, with the checkboxes on the right-hand side a la the Windows file permissions dialog.

Moreover, the current list is not sorted in any meaningful fashion. Sorting names alphabetically by first name or by last name with "Last, First" display format (Again, like the Windows file permissions dialog) would likely be much more usable.

**Comment by willdean** [comment]
I'd say that the combining the lists might not make it for the next release, but the sorting should be easily fixed.

**Comment by hmmwhatsthisdo** [comment]
What about doing "Last, First"-style display/sorting?

**Comment by willdean** [comment]
The current master displays 'Display Name', which is constructed from "GivenName Surname" - we can definitely sort that on Surname without any problem.

**Comment by hmmwhatsthisdo** [comment]
Alright. Would it also be possible to fall back to the AD displayName and/or userPrincipalName attributes if the given/surname are both blank? (One of our admin accounts only had a displayName attribute set in AD and showed up on Bonobo with a blank name)

**Comment by willdean** [comment]
The difficulty with taking DisplayName from AD at the moment would be where to store it within Bonobo, because we don't have a DisplayName field in the internal structures.  (This could be fixed at some point)

Falling back to Username (which would be the UserPrincipalName on an AD user) for DisplayName is a good idea, because showing a blank is completely hopeless.

**Comment by hmmwhatsthisdo** [comment]
Using the userPrincipalName attribute would work fine. (Perhaps, once storing the displayName attribute attribute is possible, it would be worth it to show the uPN on hovering over a username so people with multiple accounts can distinguish between them easily)

**Comment by RedX2501** [comment]
Yeah i definitely wanted to show the username when hovering but i found out no easy way of doing that with mvccheckboxlistfor do i left it for later

I thought i had sorted it by first name but maybe it got lost in refactoring. 

I'm no big fan of last, first. That is an American thing. We should make it configurable.

**Comment by willdean** [comment]
I am wondering if we do need to address the sorting issue here, because we probably used to sort 'by accident' on username, which was reasonable, and now we sort 'by accident' on GUID, which is not really very helpful.

I am included to make GetAllUsers sort on the SortName property I added in #503 - any thoughts on this?

**Comment by chilversc** [comment]
My biggest problem with the admin/contributors list is it works at the individual user level, rather than roles or team.

This is a pain if you have multiple repositories with the same set of admins or contributors and the team changes (i.e. some one leaves/joins).

**Comment by RedX2501** [comment]
@chilversc I'm confused. At least on version 5.0 there are teams that you can use for that.

**Comment by chilversc** [comment]
@RedX2501 yes, but even after assigning a team I still have to assign an administrator, and that is a specific user rather than a role.

**Comment by willdean** [comment]
@chilversc In 6.x. you won't need to assign a repo admin if you're already system administrator.  At some point in the future there will probably be some kind of 'team administrator' too.
